networkit/cpp/distance/NeighborhoodFunction.cpp

```diff
@@ -46,9 +46,10 @@ void NeighborhoodFunction::run() {
             }
         }
     }
-    for (index i = 1; i < size-1; ++i) {
-        result[i] += result[i-1];
-    }
+
+    if (result.size() > 2)
+        std::partial_sum(result.begin(), result.end() - 1, result.begin());
```

Comment:
I think the `- 1` is broken here. The previous code operated on the whole array but this replacement skips the last element.

Reply:
Ah true, the vector has size `size - 1`.